include/replace.hpp

```diff
@@ -19,12 +19,76 @@ namespace bit {
 // ========================================================================== //
 
 
-
-// Status: to do
+/*
+ * Case 0: ^10011101^
+ * Case 1: ^10011^101
+ * Case 2: 100^1110^1
+ * Case 3: 10011^101^
+ */
 template <class ForwardIt>
```

Comment:
Just a minor formatting thing (that we can discuss on monday), but when the arguments of a function are this verbose I think its best to put each on its own newline.